pkg/runtime/program.go

```diff
@@ -24,16 +26,33 @@ func NewProgram(src string, body core.Expression) (*Program, error) {
 		return nil, core.Error(core.ErrMissedArgument, "body")
 	}
 
-	return &Program{src, body}, nil
+	return &Program{src, body, params}, nil
 }
 
 func (p *Program) Source() string {
 	return p.src
 }
 
+func (p *Program) Params() []string {
+	res := make([]string, 0, len(p.params))
+
+	for name := range p.params {
```

Comment:
You can use `copy` instead
```golang
res := make([]string, len(p.params))
copy(res, p.params)
return res
```

Reply:
Always forget about ``copy`` function!

Reply:
Actually, it does not work like this. ``params`` is a map.

Reply:
Sorry, my bad